packages/core/src/indexing-store/historical.ts

```diff
@@ -41,6 +41,8 @@ export const createHistoricalIndexingStore = ({
     worker: (fn) => fn(),
   });
 
+  let event: Event | undefined;
```

Comment:
Worth considering that this pattern of "setting" the current event could be used similarly for injecting the `db`/`tx` and `client` values.